packages/jest-matchers/src/__tests__/matchers-test.js

```diff
@@ -455,3 +455,61 @@ describe('.toMatch()', () => {
     jestExpect('this?: throws').toMatch('this?: throws');
   });
 });
+
+describe('toMatchObject()', () => {
+
+  [
+    [{a: 'b', c: 'd'}, {a: 'b'}],
+    [{a: 'b', c: 'd'}, {a: 'b', c: 'd'}],
+    [{a: 'b', t: {z: 'z', x: {r: 'r'}}}, {a: 'b', t: {z: 'z'}}],
+    [{a: 'b', t: {z: 'z', x: {r: 'r'}}}, {t: {x: {r: 'r'}}}],
+    [{a: [3, 4, 5], b: 'b'}, {a: [3, 4, 5]}],
+    [{a: [3, 4, 5], b: 'b'}, {a: [3, 4]}],
```

Comment:
i'm a little scared to have the same behavior on arrays.
Properties definitely make sense, but for arrays it can be dangerous if you match some enums or other sets.

for example, this will pass, and it's pretty straightforward:

```js
expect({permissions: ['read', 'write'], ...manyOtherProps}).toMathcObject({permissions: ['read']});
```

but if the left side is a return of the function, that might be not intuitive and even confusing:
```js
expect(getPermissions()).toMathcObject({permissions: ['read']});
// i expect it to be just 'read', but there's and extra value that i don't even see, and the test is passing
```

should we match only objects, and use strict equality for arrays? what do you think?